### PR TITLE
fix: apply file accept tokens

### PR DESCRIPTION
**Pull request by ph-fritsche**
I cleaned up how the accept tokens are applied.

### CODE REVIEW

File: src/upload.js [hunk 1]
```diff
@@ -69,4 +50,21 @@ function upload(element, fileOrFiles, init) {
...
 
+function isAcceptableFile(file, accept) {
+  if (!accept) {
+    return true
+  }
+
+  const wildcards = ['audio/*', 'image/*', 'video/*']
+
+  return accept.split(',').some(acceptToken => {
+    if (accept[0] === '.') {
```
Comment (LauraBeatris) [inline src/upload.js]: Can we specify that it's an extension? Would be easier to read

```
const isExtension = accept[0] === '.'
```

File: src/upload.js [hunk 2]
```diff
@@ -69,4 +50,21 @@ function upload(element, fileOrFiles, init) {
...
+      return file.name.endsWith(accept)
+    } else if (wildcards.includes(acceptToken)) {
```
Comment (LauraBeatris) [inline src/upload.js:64]: 'audio/*'.includes('audio/mp3') - false

'audio/*' === 'audio/mp3' - false 

If it's a wildcard, we should just check the type, and not the subtype

File: src/upload.js (hunk 1 above)
Comment (ph-fritsche) [reply src/upload.js]: should read `acceptToken`.
I'll push the fix and add a comment.

File: src/upload.js (hunk 2 above)
Comment (ph-fritsche) [reply src/upload.js:64]: If the token is one of the wildcard the code checks for the type to start with the token excluding the last character.
So if the acceptToken is `audio/*` it checks if the type starts with `audio/`.

**Review by LauraBeatris — APPROVED** [review]

**Pull request merged**